runtime/src/snapshot_package.rs

```diff
@@ -192,31 +99,50 @@ pub struct SnapshotPackage {
 }
 
 impl SnapshotPackage {
-    #[allow(clippy::too_many_arguments)]
     pub fn new(
-        slot: Slot,
-        block_height: u64,
-        slot_deltas: Vec<BankSlotDelta>,
-        snapshot_links: TempDir,
-        storages: SnapshotStorages,
-        snapshot_archive_path: PathBuf,
-        hash: Hash,
+        mut accounts_package: AccountsPackage,
+        snapshot_archives_dir: PathBuf,
         archive_format: ArchiveFormat,
-        snapshot_version: SnapshotVersion,
-        snapshot_type: SnapshotType,
+        incremental_snapshot_base_slot: Option<Slot>,
     ) -> Self {
+        let (snapshot_type, snapshot_archive_path) = match incremental_snapshot_base_slot {
+            None => {
+                let snapshot_archive_path = snapshot_utils::build_full_snapshot_archive_path(
+                    snapshot_archives_dir,
+                    accounts_package.slot,
+                    &accounts_package.hash,
+                    archive_format,
+                );
+                (SnapshotType::FullSnapshot, snapshot_archive_path)
+            }
+            Some(incremental_snapshot_base_slot) => {
+                let snapshot_archive_path = snapshot_utils::build_incremental_snapshot_archive_path(
+                    snapshot_archives_dir,
+                    incremental_snapshot_base_slot,
+                    accounts_package.slot,
+                    &accounts_package.hash,
+                    archive_format,
+                );
+                snapshot_utils::filter_snapshot_storages_for_incremental_snapshot(
+                    &mut accounts_package.storages,
+                    incremental_snapshot_base_slot,
+                );
```

Comment:
Here's where the storage filtering happens for incremental snapshots.

Reply:
hmmm I feel like this is a little late in the pipeline, i.e. there was never a need for AccountsBackgroundService to waste the resources cloning out the unnecessary storages (which can be quite a few) only for them to be filtered out here. 

Seems more efficient to have the `AccountsBackgroundService` rather than the `AccountsHashVerifier` determine the incremental vs full snapshot and only clone + pass on those storages that are necessary.

Reply:
Yeah, cloning the unnecessary storages just to filter them out later is a bummer. But, the only way to fix that is to change `AccountsDb::get_snapshot_storages()`, which is where we do the cloning.

For the changes here, this just moves where the filtering is, so there isn't a change in how much gets cloned. Maybe I should open a new issue to update `AccountsDb::get_snapshot_storages()` to just return snapshot storages above a slot? What do you think?

Reply:
Yeah I think introducing an option to `get_snapshot_storages` to only clone out slots > a minimum slot is the right move. 

I think this also means `AccountsBackgroundService` should be the one setting the incremental/vs full snapshot flag here: https://github.com/solana-labs/solana/pull/19271/files#diff-387c1093824250b8106cb4207b95cf0b3ab875cd24cd1c2deed8d9fcd938c5aeR191-R205, and passing it downstream to the rest of the pipeline.

Reply:
Yeah, I agree. I think that's ultimately the right way to do this. OK. I think that means I should close out this PR. I'll get working on AccountsBackgroundService!

I'll pull out some of the changes in this PR into smaller PRs as well (like the LastFullSnapshotSlot addition to SnapshotConfig).